index.bs

```diff
@@ -669,15 +670,31 @@ HTML Processing Model: Event Loop</h4>
 
 An <a>Intersection Observer</a> processing step should take place
 during the "<i>Update the rendering</i>" steps,
-after step 10, <a>run the animation frame callbacks</a>,
+after step 12, <a>run the animation frame callbacks</a>,
 in the in the <a>HTML Processing Model</a>.
 
 This step is:
 
-<ol start="10">
-	<li>For each <a>fully active</a> {{document}} in |docs|, <a>Run the update intersection observations steps</a> for each {{IntersectionObserver}} whose {{IntersectionObserver/root}} is in the DOMtree of that {{document}}.</li>
+<ol start="13">
+	<li>For each <a>fully active</a> {{document}} in |docs|, <a>Run the update intersection observations steps</a> for that {{document}}, passing in |now| as the timestamp.</li>
 </ol>
 
+<h4 id='pending-initial-observation'>Pending initial IntersectionObserver targets</h4>
+A {{document}} is said to have <dfn>pending initial IntersectionObserver targets</dfn>
+if there is at least one {{IntersectionObserver}} meeting these criteria:
+	<ol>
+		<li>The |observer|'s {{IntersectionObserver|root}} is in the |document|
+		   (for the <a>top-level browsing context</a>, this includes <a>implicit root</a> observers).
+		</li>
+		<li>The |observer| has at least one |target| in its {{[[ObservationTargets]]}} slot for which
+		    no {{IntersectionObserverEntry}} has yet been queued.
```

Comment:
I'm not quite sure this is the right condition. This means that calling `observe()` on something that's off-screen would cause forced rendering updates all frames, right?

Reply:
Yesl; unless some of the frames could skip rendering without affecting the intersection calculation.

Reply:
Right, I think that's wrong. The "Unnecessary rendering" section already has a condition on:

> The user agent believes that updating the rendering of the Document's browsing context would have no visible effect

Which adding this other condition would break, right? That is. IntersectionObserver needs to force-update the rendering _once_, to guarantee that if the element is visible the observer runs. If the element is not visible, there's no need to do that, and you can skip rendering updates as usual (until something causes you to do that).

Reply:
I'm not sure I understand you. Here's the problematic scenario (running on a page with unchanging layout):

```
setTimeout(() => {
  let target = document.getElementById('target');
  let observer = new IntersectionObserver(entries => { alert('Hello, world!'); });
  observer.observe(target);
}, 1000);
```

In this case, it is true that "the rendering of the Document's browsing context would have no visible effect." So the implementation may skip rendering updates indefinitely, and no notification will be generated because the IntersectionObserver steps of the document lifecycle don't run if rendering is skipped. The point of the issue is that we should generate a notification for the target at the next opportunity, whether the target is visible or not.

Reply:
Ouch, yeah, sorry, I was misreading. This looks good, sorry.